src/v/cluster/rm_stm.cc

```diff
@@ -1922,7 +1922,7 @@ rm_stm::do_aborted_transactions(model::offset from, model::offset to) {
             continue;
         }
         if (_log_state.last_abort_snapshot.match(idx)) {
-            auto opt = _log_state.last_abort_snapshot;
+            auto& opt = _log_state.last_abort_snapshot;
```

Comment:
should probably be `const auto&`? before it was a value, so without going to look at `filter_interesting` having the reference be const here would be helpful.